packages/ember-data/lib/system/relationships/state/has_many.js

```diff
@@ -139,12 +139,13 @@ ManyRelationship.prototype.computeChanges = function(records) {
   for (i = 0; i < length; i++) {
     record = records[i];
     //Need to preserve the order of incoming records
-    if (hasManyArray.objectAt(i) === record ) {
+    if (hasManyArray.objectAt(i) === record) {
       continue;
     }
     this.removeCanonicalRecord(record);
     this.addCanonicalRecord(record, i);
   }
+  this.flushCanonicalLater();
```

Comment:
I honestly don't know why this is needed. I mean, this makes sense to flush the canonical state, but why later ?